src/Operations.jl

```diff
@@ -254,7 +229,7 @@ function collect_fixed!(ctx::Context, pkgs::Vector{PackageSpec}, names::Dict{UUI
 
         found_project = collect_project!(ctx, pkg, path, fix_deps_map)
         if !found_project
-            collect_require!(ctx, pkg, path, fix_deps_map)
```

Comment:
If we changed this line: https://github.com/JuliaLang/Pkg.jl/blob/b51b4b1db554cc4afc46e435eac6b010c2a2bbce/src/Operations.jl#L200 to something like
```
project_file !== nothing || pkgerror("No project file found at path `$path`.")
```
then we wouldn't have to do error handling outside of the function

Reply:
Yeah, but I wasn't sure if it ever was ok to return `nothing` from this. Will look where the function is used.

Reply:
Updated